common/rulesys.cpp

```diff
@@ -546,139 +521,83 @@ bool RuleManager::UpdateOrphanedRules(Database *db, bool quiet_update)
 
 bool RuleManager::RestoreRuleNotes(Database *db)
 {
-	std::string query("SELECT `ruleset_id`, `rule_name`, `notes` FROM `rule_values`");
-
-	auto results = db->QueryDatabase(query);
-	if (!results.Success()) {
+	const auto& l = RuleValuesRepository::All(*db);
+	if (l.empty()) {
 		return false;
 	}
 
 	int update_count = 0;
-	for (auto row = results.begin(); row != results.end(); ++row) {
-
-		auto rule = [](const char *rule_name) {
+	for (const auto& e : l) {
 
+		auto rule = [](std::string rule_name) {
 			for (auto rule_iter : s_RuleInfo) {
-				if (strcasecmp(rule_iter.name, rule_name) == 0) {
```

Comment:
This changes behavior as well

`strcasecmp `compares if strings are equal case insensitive whereas `Contains` is a partial

Reply:
Will use `Strings::Equals`.